test/unit/Reflection/ReflectionObjectTest.php

```diff
@@ -106,6 +106,19 @@ public function testGetRuntimePropertiesWithFilter() : void
         self::assertEmpty($classInfo->getProperties(\ReflectionProperty::IS_PRIVATE));
     }
 
+    public function testGetRuntimeImmediatePropertiesWithFilter() : void
+    {
+        $foo = new \StdClass();
+        $foo->bar = 'huzzah';
+
+        $classInfo = ReflectionObject::createFromInstance($foo);
+
+        self::assertEmpty($classInfo->getImmediateProperties(\ReflectionProperty::IS_STATIC));
+        self::assertCount(1, $classInfo->getImmediateProperties(\ReflectionProperty::IS_PUBLIC));
```

Comment:
Oh, this is my mistake! I thought that these examples would fail, but they don't:

https://3v4l.org/b7PUE

```php
<?php

$foo = (object) ['bar' => 'baz'];

$property = new \ReflectionProperty($foo, 'bar');

var_dump($property->getValue($foo));
```

```php
```